File: iteration1/src/course/CourseRecord.java
package iteration1.src.course;


import iteration1.src.human.Grade;

public class CourseRecord {
    //DataFields
    private Course course;
    private LetterGrade lGrade;
    private Season season;
    private Float score;
    private Grade grade;
    private Boolean isPassed;

    //Getters
    public Course getCourse() {
        return course;
    }
    public float getScore() {
        return score;
    }
    public Boolean getIsPassed() {
        return isPassed;
    }
    public Season getSeason() {
        return season;
    }
    public LetterGrade getlGrade() {
        return lGrade;
    }
    public Grade getGrade() {
        return grade;
    }
    

    //Setters
    public void setCourse(Course course) {
        this.course = course;
    }
    public void setScore(float score) {
       if(score < 0){
       // print error messages
       return;
       }
        this.score = score;
    }
    public void setIsPassed(Boolean isPassed) {
        this.isPassed = isPassed;
    }
    public void setlGrade(LetterGrade lGrade) {
        this.lGrade = lGrade;
    }
    public void setSeason(Season season) {
        this.season = season;
    }


    public CourseRecord(Course course, LetterGrade lGrade, Season season, Grade grade, Float score, Boolean isPassed){
        this.course=course;
        this.lGrade=lGrade;
        this.season=season;
        this.grade = grade;
        this.score = score;
        this.isPassed=isPassed;

    }
    //Get season details Method
    public void getFullSeasonDetails(){

    }
    

}

File: Json Serializer/CourseRecord.cs

[Serializable]
public class CourseRecord
{
    public string CourseCode { get; set; }
    public LetterGrade LetterGrade { get; set; }
    public float Grade { get; set; }
    public Season Season { get; set; }
    public Grade RecordYear { get; set; }
    public bool Passed { get; set; }

    public CourseRecord(string courseCode, LetterGrade letterGrade, float grade, Season season, Grade year, bool passed)
    {
        CourseCode = courseCode;
        LetterGrade = letterGrade;
        Grade = grade;
        Season = season;
        RecordYear = year;
        Passed = passed;
    }
}